### PR TITLE
Use premultiplied alpha for renderer clearColor

### LINKED ISSUE #3324: `background-opacity` does not function properly
Minimal reproducible config:
```
theme = catppuccin-latte
background-opacity = 0.2
```

It's especially obvious with light themes:
Image

Dark themes perform better, but the color is still not accurate:
Image

The bug is related to color blending, or more specifically, the incorrect usage of premultiplied alpha.

### PR DESCRIPTION
**Pull request by lixiaoyan**
Fixes #3324

---

I haven't tested it on Linux yet, but I believe it has the similar problem and could be fixed by this PR as well.

### CODE REVIEW

File: src/renderer/OpenGL.zig [hunk 1]
```diff
@@ -2344,9 +2344,9 @@ fn drawCellProgram(
 
     // Clear the surface
     gl.clearColor(
-        @as(f32, @floatFromInt(self.draw_background.r)) / 255,
-        @as(f32, @floatFromInt(self.draw_background.g)) / 255,
-        @as(f32, @floatFromInt(self.draw_background.b)) / 255,
+        @as(f32, @floatFromInt(self.draw_background.r)) / 255 * self.config.background_opacity,
+        @as(f32, @floatFromInt(self.draw_background.g)) / 255 * self.config.background_opacity,
+        @as(f32, @floatFromInt(self.draw_background.b)) / 255 * self.config.background_opacity,
```
Comment (gerblesh) [inline src/renderer/OpenGL.zig:2349]: Had some typing issues when trying to compile this on Linux, it expected an `f32` when an `f64` was being produced by the expression, fixed it with a `floatCast`
```suggestion
        @floatCast(@as(f32, @floatFromInt(self.draw_background.r)) / 255 * self.config.background_opacity),
        @floatCast(@as(f32, @floatFromInt(self.draw_background.g)) / 255 * self.config.background_opacity),
        @floatCast(@as(f32, @floatFromInt(self.draw_background.b)) / 255 * self.config.background_opacity),
```

**Comment by SimonYde** [comment]
I cannot get this to build on my linux machine:
```
>    └─ zig build-exe ghostty Debug native-native 1 errors
┃        > src/renderer/OpenGL.zig:2347:63: error: expected type 'f32', found 'f64'
┃        >         @as(f32, @floatFromInt(self.draw_background.r)) / 255 * self.config.background_opacit…
┃        >         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
┃        > pkg/opengl/draw.zig:5:22: note: parameter type declared here
┃        > pub fn clearColor(r: f32, g: f32, b: f32, a: f32) void {
┃        >                      ^~~
┃        > referenced by:
┃        >     drawFrame: src/renderer/OpenGL.zig:2289:13
┃        >     render: src/apprt/gtk/Surface.zig:669:35
┃        >     remaining reference traces hidden; use '-freference-trace' to see all reference traces
┃        > error: the following command failed with 1 compilation errors:
┃        > /nix/store/v9xybfsc2iyfd602s192pymyrrncni0n-zig-0.13.0/bin/zig build-exe /build/source/.zig-c…
┃        > Build Summary: 94/97 steps succeeded; 1 failed (disable with --summary none)
┃        > install transitive failure
┃        > └─ install ghostty transitive failure
┃        >    └─ zig build-exe ghostty Debug native-native 1 errors
```
Appears to be a type mismatch issue.

**Comment by gerblesh** [comment]
> I cannot get this to build on my linux machine using flake.nix:
> 
> ```
> >    └─ zig build-exe ghostty Debug native-native 1 errors
> ┃        > src/renderer/OpenGL.zig:2347:63: error: expected type 'f32', found 'f64'
> ┃        >         @as(f32, @floatFromInt(self.draw_background.r)) / 255 * self.config.background_opacit…
> ┃        >         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ┃        > pkg/opengl/draw.zig:5:22: note: parameter type declared here
> ┃        > pub fn clearColor(r: f32, g: f32, b: f32, a: f32) void {
> ┃        >                      ^~~
> ┃        > referenced by:
> ┃        >     drawFrame: src/renderer/OpenGL.zig:2289:13
> ┃        >     render: src/apprt/gtk/Surface.zig:669:35
> ┃        >     remaining reference traces hidden; use '-freference-trace' to see all reference traces
> ┃        > error: the following command failed with 1 compilation errors:
> ┃        > /nix/store/v9xybfsc2iyfd602s192pymyrrncni0n-zig-0.13.0/bin/zig build-exe /build/source/.zig-c…
> ┃        > Build Summary: 94/97 steps succeeded; 1 failed (disable with --summary none)
> ┃        > install transitive failure
> ┃        > └─ install ghostty transitive failure
> ┃        >    └─ zig build-exe ghostty Debug native-native 1 errors
> ```
> 
> Appears to be a type mismatch issue.

*Should* be fixed with the above suggestion

File: src/renderer/OpenGL.zig (hunk 1 above)
Comment (lixiaoyan) [reply src/renderer/OpenGL.zig]: Thanks!

**Comment by SimonYde** [comment]
works just fine on linux now :+1:

**Comment by yspreen** [comment]
works wonders on macOS. does fix #3324

**Pull request merged**